code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -439,7 +439,7 @@
 		return
 	if(pixel_x != base_pixel_x || pixel_y != base_pixel_y)
 		animate(src, 0.2 SECONDS, pixel_x = base_pixel_y, pixel_y = base_pixel_y, flags = ANIMATION_PARALLEL)
-	Move(get_step(src, direction), direction)
```

Comment:
why was move allowing the move between z levels? Do we really not check it at all?

Reply:
There's 2 parts to this. One is that Move doesn't check whether there's actually a link between the Z-levels you're trying to move to. The other part, which I still don't entirely understand, is that it seemed like canPass was seeing that the space tile it was trying to move to had density = 0 and just immediately returning true. I don't really understand the move, canPass, etc. chain enough to fully understand if that's actually what was happening or why. I figured since this does fix the bug, and was probably how it should've been done anyway, so this seemed like a fairly good change to make regardless of any bugs with Move itself.